tests/e2e/account/pw_register_test.py

```diff
@@ -0,0 +1,195 @@
+"""Test account registration."""
+
+import pytest
+from playwright.sync_api import Page, expect
+from pytest_bdd import given, parsers, scenarios, then, when
+
+scenarios('account/register.feature')
+pytestmark = pytest.mark.usefixtures('live_server')
+
+TWOFLOWER_EMAIL = 'twoflower@example.org'
+TWOFLOWER_PHONE = '+12015550123'
+TWOFLOWER_PASSWORD = 'te@pwd3289'  # nosec
+
+
+def check_recaptcha_loaded(page: Page) -> None:
+    page.wait_for_selector(
+        '#form-passwordlogin > div.g-recaptcha > div > div.grecaptcha-logo > iframe',
+        timeout=10000,
+    )
+
+
+@given("Anonymous visitor is on the home page")
+def given_anonuser_home_page(live_server, page: Page) -> None:
+    page.goto(live_server.url)
+    expect(page).to_have_title("Test Hasgeek")
+
+
+@when(
+    parsers.re(
+        "they navigate to the login page and submit"
+        " (?P<phone_or_email>a phone number|an email address)"
+    ),
+    target_fixture='anon_username',
+)
+def when_anonuser_navigates_login_and_submits(
+    app, live_server, phone_or_email: str, page: Page
+) -> dict[str, str]:
+    if phone_or_email == "a phone number":
+        username = '8123456789'
+    elif phone_or_email == "an email address":
+        username = 'anon@example.com'
+    else:
+        pytest.fail("Unknown username type")
+    page.click('.header__button')
+    check_recaptcha_loaded(page)
+    page.wait_for_selector('input[name=username]').fill(username)
+    page.click('#form-passwordlogin button')
+    return {'phone_or_email': phone_or_email, 'username': username}
+
+
+@then("they are prompted for their name and the OTP, which they provide")
+def then_anonuser_prompted_name_and_otp(live_server, anon_username, page: Page) -> None:
+    page.wait_for_selector('input[name=fullname]').fill('Twoflower')
+    if anon_username['phone_or_email'] == "a phone number":
+        otp = live_server.transport_calls.sms[-1].vars['otp']
+    elif anon_username['phone_or_email'] == "an email address":
+        otp = live_server.transport_calls.email[-1].subject.split(' ')[-1]
```

Comment:
For safety, this should use a regex pattern to extract the OTP from the subject. If the subject format changes, this test will fail for regex not matching instead of incorrect OTP – making the problem easier to spot.